packages/@aws-cdk/aws-codedeploy/test/integ.deployment-group.ts

```diff
@@ -0,0 +1,31 @@
+import autoscaling = require('@aws-cdk/aws-autoscaling');
+import ec2 = require('@aws-cdk/aws-ec2');
+import lb = require('@aws-cdk/aws-elasticloadbalancing');
+import cdk = require('@aws-cdk/cdk');
+import codedeploy = require('../lib');
+
+const app = new cdk.App(process.argv);
+
+const stack = new cdk.Stack(app, 'aws-cdk-codedeploy-server-dg');
+
+const vpc = new ec2.VpcNetwork(stack, 'VPC');
+
+const asg = new autoscaling.AutoScalingGroup(stack, 'ASG', {
+  instanceType: new ec2.InstanceTypePair(ec2.InstanceClass.M5, ec2.InstanceSize.Large),
+  machineImage: new ec2.AmazonLinuxImage(),
+  vpc,
+});
+
+const elb = new lb.LoadBalancer(stack, 'ELB', { vpc });
+elb.addTarget(asg);
+elb.addListener({
+  externalPort: 80,
+});
+
+new codedeploy.ServerDeploymentGroup(stack, 'CodeDeployGroup', {
+  deploymentConfig: codedeploy.ServerDeploymentConfig.AllAtOnce,
+  autoScalingGroups: [asg],
+  loadBalancer: elb,
```

Comment:
I don't like this duplication very much. (2/2)

We're both saying that the ASG should be a load balancer target (directly), AND that the ASG is a load balancer target (via the DeploymentGroup).

Why are both necessary? I mean, I can think of some mechanical reasons why this is necessary today: probably the DeploymentGroup is going to call `register-instance` and `register-instance` only when it makes changes, and the direct association between load balancer and ASG is necessary to register the *initial* instances, before CodeDeploy makes any changes... but I still don't really like it.

Can't we interrogate the ASGs for their load balancers and copy those onto the DeploymentGroup?

Reply:
> Why are both necessary?

Honestly, I'm not even 100% sure that this duplication is required. Perhaps setting the ASG just on the Deployment Group, and then setting the load balancer on the Deployment Group, is actually enough? I'll do some research.

> Can't we interrogate the ASGs for their load balancers and copy those onto the DeploymentGroup?

That won't work. You can have a load balancer for a Deployment Group without any ASGs (when the Deployment Group uses tags for EC2/on-premise instances).

Reply:
> That won't work. You can have a load balancer for a Deployment Group without any ASGs (when the Deployment Group uses tags for EC2/on-premise instances).

One doesn't have to preclude the other. I want to (be able to) get rid of the repetition if it exists.

Also, you could still have an object that represents the tagged instances.